examples/Misc/NavigateSearchResultPagesCachingTokens.cs

```diff
@@ -0,0 +1,265 @@
+    /// </summary>
+    public class NavigateSearchResultPagesCachingTokens : ExampleBase
+    {
+        /// <summary>
+        /// Command line options for running the <see cref="NavigateSearchResultPagesCachingTokens"/> example.
+        /// </summary>
+        public class Options : OptionsBase
+        {
+            /// <summary>
+            /// The Google Ads customer ID for which the call is made.
+            /// </summary>
+            [Option("customerId", Required = true, HelpText =
+                "The Google Ads customer ID for which the call is made.")]
+            public long CustomerId { get; set; }
+        }
+
+        /// <summary>
+        /// The maximum number of results to retrieve.
+        /// </summary>
+        private const long RESULTS_LIMIT = 11;
+
+        /// <summary>
+        /// The size of the paginated search result pages.
+        /// </summary>
+        private const int PAGE_SIZE = 3;
+
+        /// <summary>
+        /// Main method, to run this code example as a standalone application.
+        /// </summary>
+        /// <param name="args">The command line arguments.</param>
+        public static void Main(string[] args)
+        {
+            Options options = new Options();
+            CommandLine.Parser.Default.ParseArguments<Options>(args).MapResult(
+                delegate (Options o)
+                {
+                    options = o;
+                    return 0;
+                }, delegate (IEnumerable<Error> errors)
+                {
+                    // The Google Ads customer ID for which the call is made.
+                    options.CustomerId = long.Parse("INSERT_CUSTOMER_ID_HERE");
+                    return 0;
+                }
+            );
+            NavigateSearchResultPagesCachingTokens codeExample =
+                new NavigateSearchResultPagesCachingTokens();
+            Console.WriteLine(codeExample.Description);
+            codeExample.Run(new GoogleAdsClient(), options.CustomerId);
+        }
+
+        /// <summary>
+        /// Returns a description about the code example.
+        /// </summary>
+        public override string Description => "GoogleAdsService.Search results are paginated but " +
+            "they can only be retrieved in sequence starting by the first page. More details " +
+            "at https://developers.google.com/google-ads/api/docs/reporting/paging. This code " +
+            "example searches campaigns illustrating how GoogleAdsService.Search result page " +
+            "tokens can be cached and reused to retrieve previous pages.This is useful when " +
+            "you need to request pages that were already requested in the past without starting " +
+            "over from the first page.For example, it can be used to implement an interactive " +
+            "application that displays a page of results at a time without caching all the " +
+            "results first.";
```

Comment:
Missing space after periods in this block as well

Reply:
Done.